packages/paste-core/components/tabs/__tests__/tabs.test.tsx

```diff
@@ -92,6 +97,317 @@ describe('Tabs', () => {
+            <Tab data-testid="tab-1">Tab 1 is a long tab name because the server sent a long tab name</Tab>
+            <Tab data-testid="tab-2">Tab 2</Tab>
+            <Tab data-testid="tab-3">Tab 3</Tab>
+          </TabList>
+          <TabPanels data-testid="tab-panels">
+            <TabPanel data-testid="tab-panel-1">Tab 1</TabPanel>
+            <TabPanel data-testid="tab-panel-2">Tab 2</TabPanel>
+            <TabPanel data-testid="tab-panel-3">Tab 3</TabPanel>
+          </TabPanels>
+        </Tabs>
+      );
+
+      const outerDiv = screen.getByTestId('tab-list').parentElement as HTMLElement;
+
+      expect(outerDiv.getAttribute('data-paste-element')).toEqual('VERTICAL_TABS');
+      expect(screen.getByTestId('tab-list').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB_LIST');
+      expect((screen.getByTestId('tab-list').firstChild as HTMLElement).getAttribute('data-paste-element')).toEqual(
+        'VERTICAL_TAB_LIST_CHILD'
+      );
+      expect(screen.getByTestId('tab-1').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB');
+      expect(screen.getByTestId('tab-2').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB');
+      expect(screen.getByTestId('tab-3').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB');
+      expect(screen.getByTestId('tab-panels').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB_PANELS');
+      expect(screen.getByTestId('tab-panel-1').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB_PANEL');
+      expect(screen.getByTestId('tab-panel-2').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB_PANEL');
+      expect(screen.getByTestId('tab-panel-3').getAttribute('data-paste-element')).toEqual('VERTICAL_TAB_PANEL');
+    });
+
+    it('should set an element data attribute Horizontal Tabs', () => {
+      render(
+        <Tabs element="HORSE" orientation="horizontal" baseId="">
+          <TabList element="CAT" data-testid="tab-list" aria-label="My tabs">
+            <Tab element="TIGER" data-testid="tab-1">
+              Tab 1 is a long tab name because the server sent a long tab name
+            </Tab>
+            <Tab element="PANTHER" data-testid="tab-2">
+              Tab 2
+            </Tab>
+            <Tab element="SCOTTISH_FOLD" data-testid="tab-3">
+              Tab 3
+            </Tab>
+          </TabList>
+          <TabPanels element="DOG" data-testid="tab-panels">
+            <TabPanel element="CORGI" data-testid="tab-panel-1">
+              Tab 1
+            </TabPanel>
+            <TabPanel element="GOLDEN_DOODLE" data-testid="tab-panel-2">
+              Tab 2
+            </TabPanel>
+            <TabPanel element="PUG" data-testid="tab-panel-3">
+              Tab 3
+            </TabPanel>
+          </TabPanels>
+        </Tabs>
+      );
+
+      const outerDiv = screen.getByTestId('tab-list').parentElement as HTMLElement;
+
+      expect(outerDiv.getAttribute('data-paste-element')).toEqual('HORIZONTAL_HORSE');
```

Comment:
I don't think we should prefix the direction for custom elements because it breaks the convention of element overwriting defaults and may come as a frustrating surprise to developers. Curious about your rationale though